src/arch/powerpc64.rs

```diff
@@ -0,0 +1,702 @@
+//! Low-level PowerPC support.
+//!
+//! This file is heavily based on the x86_64 implementation.
+//! Relevant differences are highlighted in comments, but otherwise most
+//! comments have been removed to avoid duplication. Refer to x86_64.rs for
+//! detailed comments about what is happening in this file.
+//!
+//! ## Stack layout
+//!
+//! Here is what the layout of the stack looks like when a coroutine is
+//! suspended.
+//!
+//! Note: "link area" refers to the minimal 32-byte stack frame header that is
+//! required by the PowerPC ABI.
+//!
+//! ```text
+//! +--------------+  <- Stack base
+//! | Padding      |  <-
+//! +--------------+   |
+//! | Initial func |  <-
+//! +--------------+   | This mirrors the 32-byte link area layout.
+//! | Padding      |  <-
+//! +--------------+   |
+//! | Parent link  |  <-
+//! +--------------+ <- The 4 fields above form a fake link area which
+//! |              |    backchains back to the parent stack.
+//! ~     ...      ~
+//! |              |
+//! +--------------+
+//! |              |
+//! | Link area    |  <- 32-byte link area of the last frame before the suspend.
+//! |              |     The LR field holds the PC to resume execution at.
+//! +--------------+  <- Saved stack pointer points here.
+//! | Saved R31    |
+//! +--------------+
+//! | Saved R30    |
+//! +--------------+
+//! | Saved R29    |
```

Comment:
As of https://github.com/rust-lang/rust/pull/148492 r29 can just be specified as a clobber, we don't need to preserve it here.